### PR TITLE
Align Java DSL package structure with Scala DSL

### LINKED ISSUE #20945: Align package structure in HTTP javadsl
Some classes in JavaDSL live inside `akka.http.javadsl.server` while they should live outside in `akka.http.javadsl.marshalling / unmarshalling / common`.

All classes exist as far as I see, they just should be moved in the semantically right package before we go stable.

### PR DESCRIPTION
**Pull request by johanandren**
Supersedes #20966,

Fixes #20945

### CODE REVIEW

File: akka-docs/rst/scala/http/migration-guide-2.4.x-experimental.rst [hunk 1]
```diff
@@ -24,3 +24,19 @@ Akka HTTP 2.4.7 -> 2.4.8
...
+Akka HTTP 2.4.8 -> 2.4.9
+------------------------
+
+Java DSL Package structure changes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+We have aligned the package structure of the Java based DSL with the Scala based DSL
+and moved classes that was in the wrong or unexpected places around a bit. This means
+that Java DSL users must update their imports as follows:
+
+Classes dealing with unmarshalling and unmarshalling used to reside in `akka.http.javadsl.server`
```
Comment (ktoso) [inline akka-docs/rst/scala/http/migration-guide-2.4.x-experimental.rst]:
```
``
```  

because rst

**Comment by ktoso** [comment]
LGTM, let's actually merge it now, I'll rebase my things

File: akka-docs/rst/scala/http/migration-guide-2.4.x-experimental.rst [hunk 2]
```diff
@@ -24,3 +24,19 @@ Akka HTTP 2.4.7 -> 2.4.8
...
 thus it was moved to become such. It is now available as `akka.http.javadsl.model.headers.HttpChallenge#create[Basic|OAuth2]`
 for JavaDSL and `akka.http.scaladsl.model.headers.HttpChallenges#[basic|oAuth2]` for ScalaDSL.
+
+Akka HTTP 2.4.8 -> 2.4.9
+------------------------
+
+Java DSL Package structure changes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+We have aligned the package structure of the Java based DSL with the Scala based DSL
+and moved classes that was in the wrong or unexpected places around a bit. This means
```
Comment (2beaucoup) [inline akka-docs/rst/scala/http/migration-guide-2.4.x-experimental.rst]: "that were"

File: akka-docs/rst/scala/http/migration-guide-2.4.x-experimental.rst (hunk 1 above)
Comment (johanandren) [inline akka-docs/rst/scala/http/migration-guide-2.4.x-experimental.rst]: damned markup languages

Comment (2beaucoup) [inline akka-docs/rst/scala/http/migration-guide-2.4.x-experimental.rst]: "unmarshalling and unmarshalling"

**Comment by johanandren** [comment]
Hmm. I'll just link the Scala one from the java doc side.

File: akka-docs/rst/scala/http/migration-guide-2.4.x-experimental.rst [hunk 3]
```diff
@@ -19,8 +19,24 @@ versions of the **experimental** part of Akka HTTP.
...
+that Java DSL users must update their imports as follows:
+
+Classes dealing with unmarshalling and unmarshalling used to reside in ``akka.http.javadsl.server``
```
Comment (2beaucoup) [inline akka-docs/rst/scala/http/migration-guide-2.4.x-experimental.rst]: "unmarshalling and unmarshalling"

**Comment by ktoso** [comment]
Yeah the link can work for now I guess.

**Comment by ktoso** [comment]
LGTM, please merge once validated

File: akka-docs/rst/scala/http/migration-guide-2.4.x-experimental.rst [hunk 4]
```diff
@@ -19,8 +19,24 @@ versions of the **experimental** part of Akka HTTP.
...
+Akka HTTP 2.4.8 -> 2.4.9
+------------------------
+
+Java DSL Package structure changes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+We have aligned the package structure of the Java based DSL with the Scala based DSL
+and moved classes that were in the wrong or unexpected places around a bit. This means
+that Java DSL users must update their imports as follows:
+
+Classes dealing with unmarshalling and unmarshalling used to reside in ``akka.http.javadsl.server``
```
Comment (2beaucoup) [inline akka-docs/rst/scala/http/migration-guide-2.4.x-experimental.rst]: "unmarshalling and unmarshalling" ;)

Comment (ktoso) [inline akka-docs/rst/scala/http/migration-guide-2.4.x-experimental.rst]: ^_^''

**Pull request merged**